telemetrylib/SQL.cpp

```diff
@@ -0,0 +1,129 @@
+        finish = true;
+        t->join();
+    }
+
+    void sendData(QByteArray bytes) override {
+        auto curr_msec = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+
+        QUrl myurl;
+        myurl.setScheme("http");
+        myurl.setHost("150.136.104.125"); // TODO
+        myurl.setPort(3000); // TODO
+        myurl.setPath("/add-data"); // TODO
+        myurl.setQuery("table-name=" + tableName + "&dataset-time=" + QString::fromStdString(std::to_string(curr_msec)));
+
+        //QNetworkRequest request;
+        request.setUrl(myurl);
+        request.setHeader(QNetworkRequest::ContentTypeHeader, QVariant("arraybuffer")); // TODO Try "blob" for content type as well
+        request.setAttribute(QNetworkRequest::HttpPipeliningAllowedAttribute, true);
+        this->restclient->post(request, bytes);
+    }
+
+    /*
+     std::string receiveData() override{
+        QJsonObject json = QJsonDocument::fromJson(reply->readAll()).object();
+        if (json.isEmpty()) {
+            qDebug() << "EMPTY JSON";
+            return "nada";
+        } else {
+            tableName = json.take("response").toString();
+            qDebug() << "HTTP response (table name): " << tableName;
+            return tableName.toStdString();
+            // TODO Automatically delete server responses since they aren't used after reading the table name
+            this->restclient->setAutoDeleteReplies(true);
+        }
+    }
+    */
+
+    bool getConnectionStatus() override{
+        // Send request to create a new table when connection to server is first established
+        if(tableName.isNull() && connection) {
+            qDebug() << "Requested a new table: " << tableToCreate;
+
+            QUrl myurl;
+            myurl.setScheme("http");
+            myurl.setHost("150.136.104.125"); 
+            myurl.setPort(3000);
+            myurl.setPath("/add-table/" + tableToCreate);
+
+            request.setUrl(myurl);
+            reply = restclient->get(request);
+
+            connect(reply, &QNetworkReply::readyRead, this, &SQL::readReply);
+            request.setHeader(QNetworkRequest::ContentTypeHeader, QVariant("arraybuffer"));
+            request.setAttribute(QNetworkRequest::HttpPipeliningAllowedAttribute, true);
+        }
+
+        return connection;
+    }
+
+
+public slots:
+    /* Not doing this for now
+    void readReply() override{
+        qDebug()<<"read reply invoked";
+        QJsonObject json = QJsonDocument::fromJson(reply->readAll()).object();
+
+        if(json.isEmpty()) {
+            qDebug() << "EMPTY JSON";
+        } else {
+            tableName = json.take("response").toString();
+            qDebug() << "HTTP response (table name): " << tableName;
+
+            // TODO Automatically delete server responses since they aren't used after reading the table name
+            this->restclient->setAutoDeleteReplies(true);
+        }
+    }
+    */
```

Comment:
```suggestion
    /**
     * Read response from the server. Specifically, reads the response to the request to
     * add a new table on the server and sets tableName to the response.
     */
    void readReply() override {
        qDebug()<<"read reply invoked";
        QJsonObject json = QJsonDocument::fromJson(reply->readAll()).object();

        if(json.isEmpty()) {
            qDebug() << "EMPTY JSON";
        } else {
            tableName = json.take("response").toString();
            qDebug() << "HTTP response (table name): " << tableName;

            // TODO Automatically delete server responses since they aren't used after reading the table name
            this->restclient->setAutoDeleteReplies(true);
        }
    }
```
**readReply() has to be included.** This is used to get the table name from the server.
**Also, added a function header.**